sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3086,6 +3086,29 @@ object GpuOverrides extends Logging {
          |For instance decimal strings not longer than 18 characters / hexadecimal strings
          |not longer than 15 characters disregarding the sign cannot cause an overflow.
          """.stripMargin.replaceAll("\n", " ")),
+    expr[FormatNumber](
+      "Formats the number x like '#,###,###.##', rounded to d decimal places.",
+      ExprChecks.binaryProject(TypeSig.STRING, TypeSig.STRING,
+        ("x", TypeSig.gpuNumeric, TypeSig.cpuNumeric),
```

Comment:
We may need to disable DecimalType support temporarily due to the bug https://github.com/NVIDIA/spark-rapids/issues/9309

Reply:
If you agree, I'd like to update it in a follow-on pr (maybe disable it along with `round`) if this pr is going to be merged.

Reply:
ok, but usually need an issue to track this followup.

Reply:
filed #9322

Reply:
I would prefer to add it in this PR, since we already know there is an issue for DecimalType.

Reply:
updated.